config/management/src/genesis.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{constants, error::Error, SingleBackend};
+use libra_secure_storage::Storage;
+use libra_types::transaction::{Transaction, TransactionPayload};
+use std::{convert::TryInto, path::PathBuf};
+use structopt::StructOpt;
+
+// TODO(davidiw) add operator_address, since that will eventually be the identity producing this.
+#[derive(Debug, StructOpt)]
+pub struct Genesis {
+    #[structopt(flatten)]
+    backend: SingleBackend,
+    #[structopt(long)]
+    path: Option<PathBuf>,
+}
+
+impl Genesis {
+    pub fn execute(self) -> Result<Transaction, Error> {
+        // Step 1) Retrieve layout
+        let mut common_config = self.backend.backend.clone();
```

Comment:
Can these steps not be refactored into individual helper methods (or something similar)? I find these long methods difficult to reason about.. It's hard to know what goes out of scope. Plus it also seems there are parts of each step  which are pretty self contained.